app/renderer/components/Inspector/HighlighterRects.js

```diff
@@ -1,20 +1,152 @@
 import React, { Component } from 'react';
 import HighlighterRect from './HighlighterRect';
+import HighlighterCentroid from './HighlighterCentroid';
 import B from 'bluebird';
-import { SCREENSHOT_INTERACTION_MODE } from './shared';
+import { SCREENSHOT_INTERACTION_MODE, parseCoordinates } from './shared';
 
 const {TAP, SWIPE, SELECT} = SCREENSHOT_INTERACTION_MODE;
+const RENDER = {CENTROID: 'centroid', EXPAND: 'expand', OVERLAP: 'overlap'};
 
 /**
  * Shows screenshot of running application and divs that highlight the elements' bounding boxes
  */
 export default class HighlighterRects extends Component {
 
+  getElements (source) {
+    const elementsByOverlap = this.findElementProps(source, null, [], {});
+    let elements = [];
+
+    for (let key in elementsByOverlap) {
+      let element = elementsByOverlap[key][0];
+
+      //Create new element obj which will be a +/- centroid
+      if (elementsByOverlap[key].length > 1) {
+        element = {
+          type: RENDER.EXPAND,
+          element: null,
+          parent: null,
+          properties: {
+            left: null,
+            top: null,
+            width: null,
+            height: null,
+            centerX: element.properties.centerX,
+            centerY: element.properties.centerY,
+            path: key,
+            keyCode: key,
+            container: null,
+            accessible: null
+          }
+        };
+        elements = elements.concat(this.newCoordinates(elementsByOverlap[key], key));
+      }
+      elements.push(element);
+    }
+
+    // Remove first element as it is an XML document node
+    elements.shift();
+
+    return elements;
+  }
+
+  // Create a new object for each element and determine its properties
+  findElementProps (element, prevElement, elements, overlaps) {
+    if (!element) {
+      return {};
+    }
+    const { scaleRatio, containerEl } = this.props;
+    const {x1, y1, x2, y2} = parseCoordinates(element);
+    const xOffset = this.highlighterXOffset || 0;
+    const posOffset = 0.03 * containerEl.offsetWidth;
+    const obj = {
+      type: RENDER.CENTROID,
+      element,
+      parent: prevElement,
+      properties: {
+        left: x1 / scaleRatio + xOffset || 0,
+        top: y1 / scaleRatio || 0,
+        width: (x2 - x1) / scaleRatio || 0,
+        height: (y2 - y1) / scaleRatio || 0,
+        centerX: ((Math.round(x1 + ((x2 - x1) / 2)) / scaleRatio) + xOffset - posOffset) || 0,
```

Comment:
if you create intermediate variables for left, top, width, height, you might be able to define centerpoints more easily, without as much repetition of math